jobs/build/ocp/Jenkinsfile

```diff
@@ -147,6 +153,46 @@ ${OA_CHANGELOG}
     }
 }
 
+// extract timing information from the record_log and write a report string
+// the timing record log entry has this form:
+// image_build_metrics|elapsed_total_minutes={d}|task_count={d}|elapsed_wait_minutes={d}|
+def get_build_timing_report() {
+
+    record_log = buildlib.parse_record_log(OIT_WORKING)
+    metrics = record_log['image_build_metrics']
+    if metrics == null || metrics.size == 0 {
+	return ""
+    }
+
+    // metrics[0]['task_count']
+    // metrics[0]['elapsed_total_minutes']
+    // metrics[0]['elapsed_wait_minutes']
+
+    return """
+Images built: ${metrics[0]['task_count']}
+Elapsed image build time: ${metrics[0]['elapsed_total_minutes']} minutes
+Time spent waiting for OSBS capacity: ${metrics[0]['elapsed_wait_minutes']} minutes
+""".trim()
+}
+
+// get the list of images built
+def get_image_build_report() {
+
+    record_log = buildlib.parse_record_log(OIT_WORKING)
+    builds = record_log['build']
+    image_list = ["Images included in build:"]
+
+    for (i = 0; i < builds.size(); i++) {
+        bld = builds[i]
+        if bld['status'] == "0" && bld['push_status'] == "0" {
+            image_spec_string =
+                "${bld['image']:${bld['version']}-${bld['release']}"
+            image_list << image_spec_string
+        }
+    }
+    return imagelist.join("\n    ")
```

Comment:
First image would not be indented. You can install groovy locally and try out code in groovyConsole to as a preview (i.e. you can run well defined snippets with pre-populated input to check syntax/etc).

Reply:
yes it would because I seeded the list with the header string and joined with \n\s{3}